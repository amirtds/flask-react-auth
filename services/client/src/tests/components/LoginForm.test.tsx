import React from "react";
import { render, screen } from "@testing-library/react";
import { it, expect } from "vitest";
import "@testing-library/jest-dom/vitest";
import LoginForm from "../../components/LoginForm";

it("LoginForm renders without crashing", () => {
  render(<LoginForm onSubmit={() => {}} />);

  expect(screen.getByRole("heading", { name: /log in/i })).toBeInTheDocument();
  expect(screen.getByLabelText(/email/i)).toBeInTheDocument();
  expect(screen.getByLabelText(/password/i)).toBeInTheDocument();
  expect(screen.getByRole("button", { name: /log in/i })).toBeInTheDocument();
});

it("LoginForm renders properly", () => {
  const { asFragment } = render(<LoginForm onSubmit={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
