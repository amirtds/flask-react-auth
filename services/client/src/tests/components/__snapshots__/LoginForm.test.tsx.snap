// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginForm renders properly 1`] = `
<DocumentFragment>
  <div
    class="css-1iq6qq8"
  >
    <h1
      class="chakra-heading css-ov0hf1"
    >
      Log In
    </h1>
    <form>
      <div
        class="chakra-stack css-1xb2ccj"
      >
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            for="email"
            id="email-label"
          >
            Email
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <input
            aria-required="true"
            class="chakra-input css-0"
            id="email"
            name="email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            for="password"
            id="password-label"
          >
            Password
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <input
            aria-required="true"
            class="chakra-input css-0"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
        <button
          class="chakra-button css-l01cow"
          type="submit"
        >
          Log In
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
